api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1650,4 +1650,13 @@ private void withCurrentUserAs(User user, Runnable internals) throws IllegalAcce
 			FieldUtils.getField(UserContext.class, "user", true).set(userContext, authenticatedUser);
 		}
 	}
+
+	@Test
+	public void saveUserProperty_shouldAddANewPropertyWithAVeryLargeStringWithoutRunningIntoError() {
+		final String USER_PROPERTY_KEY = "emrapi.lastViewedPatientIds";
+		final String USER_PROPERTY_VALUE = liquibase.util.StringUtil.repeat("1234",500);
+		// save a new user with similar properties and compare 
+		User updatedUser = userService.saveUserProperty(USER_PROPERTY_KEY, USER_PROPERTY_VALUE);
+		assertNotNull(updatedUser.getUserProperties());
```

Comment:
What if it has just been silently truncated and saved?

Reply:
@dkayiwa i have done it this way. At UserServiceImpl api level . Checking docs , postgres throws an error by default .Mysql / MariaDB need to be configured .

Reply:
I did not mean to make any other changes elsewhere. All i wanted to say was that simply asserting that user properties is not null does not mean that it contains the properties that you actually saved. You can have a an empty non null collection. Or one with values different from those that you saved.

Reply:
Ooh ,thanks so if am to get this clear ,is their any validation check you want me to add at this point ?